rke_tests/common.py

```diff
@@ -0,0 +1,214 @@
+    """
+    Creates a daemonset of pods, one pod per worker node
+    Gets pod name, pod ip, host ip, and containers
+    For each pod, use kubectl exec to ping all other pod ips
+    Asserts that each ping is successful
+    Tears down daemonset
+    """
+
+    namespace = 'default'
+    yml_file = 'resources/k8s_ymls/daemonset_test1.yaml'
+
+    # create pod on each worker node
+    result = kubectl.create_resourse_from_yml(yml_file, namespace=namespace)
+    assert result.ok, result.stderr
+
+    kubectl.wait_for_pods(
+        selector='name=daemonset-test1', namespace=namespace)
+    # get pods on each node/namespaces to test intercommunication with pods on
+    # different nodes
+    pods = kubectl.get_resource(
+        'pods', namespace=namespace, selector='name=daemonset-test1')
+    pod_list = {}
+    for pod in pods['items']:
+        pod_name = pod['metadata']['name']
+        pod_list[pod_name] = {
+            'pod_ip': pod['status']['podIP'],
+            'host_ip': pod['status']['hostIP'],
+            'containers': [x['name'] for x in pod['spec']['containers']]}
+
+    for pod_name in pod_list.keys():
+        for pod, pod_info in pod_list.iteritems():
+            if pod_name == pod:
+                continue  # Skip pinging self
+            cmd = 'ping -c 1 {0}'.format(pod_info['pod_ip'])
+            result = kubectl.exec_cmd(pod_name, cmd, namespace)
+            expect_result = '1 packets transmitted, 1 received, 0% packet loss'
+            assert expect_result in result.stdout
+
+    result = kubectl.delete_resourse_from_yml(yml_file, namespace=namespace)
+
+
+def validate_dns_service_discovery(kubectl):
+    """
+    Creates two namespaces
+    Creates a service in each namespace, gets service IP
+    Creates a single pod in first namespace
+    With pod, service dns record,
+    kubectl exec dig dns_record +short and assert that service IP
+    """
+
+    namespace_one = 'nsone'
+    namespace_two = 'nstwo'
+    services = {
+        'k8test1': {
+            'namespace': namespace_one,
+            'yml_file': 'resources/k8s_ymls/service1_ingress.yml',
+        },
+        'k8test2': {
+            'namespace': namespace_two,
+            'yml_file': 'resources/k8s_ymls/service2_ingress.yml',
+        }
+    }
+
+    kubectl.create_ns(namespace_one)
+    kubectl.create_ns(namespace_two)
+
+    dns_records = {}
+    for service_name, service_info in services.iteritems():
+        namespace = service_info['namespace']
+        # create service
+        result = kubectl.create_resourse_from_yml(
+            service_info['yml_file'], namespace=namespace)
+        assert result.ok, result.stderr
+        # map expected IP to dns service name
+        dns = "{0}.{1}.svc.cluster.local".format(service_name, namespace)
+        svc = kubectl.get_resource(
+            'svc', resource_name=service_name, namespace=namespace)
+        dns_records.update({dns: svc["spec"]["clusterIP"]})
+
```

Comment:
Can we actually make a check to connect using the dns name and see it resolve to one of the pods in the service?

Reply:
Split creation and validation as 2 function calls to make them usable independently and make namespace a parameter to these methods.